test/e2e/servicedns.go

```diff
@@ -90,60 +92,147 @@ var _ = Describe("MultiClusterServiceDNS", func() {
+		It("ServiceDNS object status should be updated, when service and endpoint are created in member cluster", func() {
+			namespace := f.TestNamespaceName()
 
-		serviceDNSStatus := dnsv1a1.MultiClusterServiceDNSRecordStatus{DNS: []dnsv1a1.ClusterDNS{}}
+			By("Create the ServiceDNS object")
+			serviceDNSObj := common.NewServiceDNSObject(baseName, namespace)
+			serviceDNS, err := fedClient.MulticlusterdnsV1alpha1().MultiClusterServiceDNSRecords(namespace).Create(serviceDNSObj)
+			framework.ExpectNoError(err, "Error creating MultiClusterServiceDNS object %v", serviceDNS)
+			name := serviceDNS.Name
+
+			serviceDNSStatus := &dnsv1a1.MultiClusterServiceDNSRecordStatus{DNS: []dnsv1a1.ClusterDNS{}}
+
+			By("Create corresponding service and endpoint for the ServiceDNS object in member clusters")
+			serviceDNSStatus = createClusterServiceAndEndpoints(f, name, namespace, serviceDNSStatus)
+
+			serviceDNS.Status = *serviceDNSStatus
+
+			By("Wait for the ServiceDNS object to have correct status")
+			common.WaitForObject(tl, namespace, name, objectGetter, serviceDNS, framework.PollInterval, wait.ForeverTestTimeout)
+		})
+
+		It("DNSEndpoint object should be created with correct status when ServiceDNS object is created", func() {
+			namespace := f.TestNamespaceName()
+
+			By("Create the ServiceDNS object")
+			serviceDNSObj := common.NewServiceDNSObject(baseName, namespace)
+			serviceDNSObj.Spec.FederationName = federation
+			serviceDNSObj.Spec.DNSSuffix = dnsZone
+			serviceDNS, err := fedClient.MulticlusterdnsV1alpha1().MultiClusterServiceDNSRecords(namespace).Create(serviceDNSObj)
+			framework.ExpectNoError(err, "Error creating MultiClusterServiceDNS object %v", serviceDNS)
+			name := serviceDNS.Name
 
-		service := common.NewServiceObject(name, namespace)
-		endpoint := common.NewEndpointObject(name, namespace)
-		lbsuffix := 1
+			serviceDNSStatus := &dnsv1a1.MultiClusterServiceDNSRecordStatus{DNS: []dnsv1a1.ClusterDNS{}}
 
-		By("Create corresponding service and endpoint for the ServiceDNS object in member clusters")
-		for clusterName, client := range f.ClusterKubeClients(userAgent) {
-			clusterLb := fmt.Sprintf("10.20.30.%d", lbsuffix)
-			lbsuffix++
+			By("Create corresponding service and endpoint for the ServiceDNS object in member clusters")
+			serviceDNSStatus = createClusterServiceAndEndpoints(f, name, namespace, serviceDNSStatus)
 
-			loadbalancerStatus := apiv1.LoadBalancerStatus{Ingress: []apiv1.LoadBalancerIngress{{IP: clusterLb}}}
-			serviceDNSStatus.DNS = append(serviceDNSStatus.DNS, dnsv1a1.ClusterDNS{Cluster: clusterName, LoadBalancer: loadbalancerStatus})
+			serviceDNS.Status = *serviceDNSStatus
 
-			// Ensure the test namespace exists in the target cluster.
-			_, err = client.CoreV1().Namespaces().Create(&apiv1.Namespace{
+			By("Wait for the ServiceDNS object to have correct status")
+			common.WaitForObject(tl, namespace, name, objectGetter, serviceDNS, framework.PollInterval, wait.ForeverTestTimeout)
```

Comment:
Am I missing something, or does the first part of this test duplicate the previous test?  If so, consider removing the previous test.

Reply:
yes, you were right except for very small difference, but it does not make much impact so removing the earlier test case.